File: projects/store-or/src/test/java/com/robinfood/storeor/usecase/getconfigurationposbystore/GetConfigurationPosByStoreUseCaseTest.java
package com.robinfood.storeor.usecase.getconfigurationposbystore;

import com.robinfood.storeor.dtos.configurationposbystore.StorePosDTO;
import com.robinfood.storeor.entities.APIResponseEntity;
import com.robinfood.storeor.entities.configurationposbystore.StorePosEntity;
import com.robinfood.storeor.entities.token.TokenModel;
import com.robinfood.storeor.mappers.IStorePosMapper;
import com.robinfood.storeor.mocks.entity.configurationposbystore.StorePosEntityMock;
import com.robinfood.storeor.repositories.configurationposbystorerepository.IStorePosRepository;
import com.robinfood.storeor.usecase.gettokenbusinesscapability.IGetTokenBusinessCapabilityUseCase;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.mockito.Mockito.when;

@ExtendWith(MockitoExtension.class)
class GetConfigurationPosByStoreUseCaseTest {

    @Mock
    private IGetTokenBusinessCapabilityUseCase getTokenBusinessCapabilityUseCase;

    @Mock
    private IStorePosRepository storePosRepository;

    @InjectMocks
    private GetConfigurationPosByStoreUseCase getConfigurationPosByStoreUseCase;

    private final String token = "token";

    final List<StorePosEntity> storePosEntitiesMocks = new StorePosEntityMock().storePosEntities;

    APIResponseEntity<List<StorePosEntity>> apiResponse = new APIResponseEntity<>(
            200, storePosEntitiesMocks,
            "2022-12-13T21:31:57.837443Z",
            "Configuration Pos by Store successfully",
            "OK"
    );

    @Mock
    private IStorePosMapper storePosMapper;

    @Test
    void test_invoke_Returns_Correctly() {

        when(getTokenBusinessCapabilityUseCase.invoke()).thenReturn(TokenModel.builder()
                .accessToken(token)
                .build()
        );

        when(storePosRepository.getConfigurationPosByStore(1L, token)).thenReturn(apiResponse);

        List<StorePosDTO> storePosDTOS = getConfigurationPosByStoreUseCase.invoke(1L);

        assertEquals(storePosMapper.storePosEntitiesToStorePosDTOs(apiResponse.getData()), storePosDTOS);
    }
}

File: projects/posv2/app/src/test/kotlin/com/robinfood/app/usecases/getposrelatedtoastore/GetConfigurationPosByStoreUseCaseTest.kt
package com.robinfood.app.usecases.getposrelatedtoastore

import com.robinfood.app.mocks.dailyreportevoucher.StorePosDTOMock
import com.robinfood.app.mocks.witnesstape.StoreOrderRequestDTOMocks
import com.robinfood.core.enums.Result
import com.robinfood.repository.posrelatedtoastore.ConfigurationPosByStoreRepository
import kotlinx.coroutines.runBlocking
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows
import org.junit.jupiter.api.extension.ExtendWith
import org.mockito.InjectMocks
import org.mockito.Mock
import org.mockito.Mockito
import org.mockito.junit.jupiter.MockitoExtension
import org.springframework.http.HttpStatus
import org.springframework.web.server.ResponseStatusException

@ExtendWith(MockitoExtension::class)
class GetConfigurationPosByStoreUseCaseTest {

    @Mock
    private lateinit var configurationPosByStoreRepository: ConfigurationPosByStoreRepository

    @InjectMocks
    private lateinit var getConfigurationPosByStoreUseCase: GetConfigurationPosByStoreUseCase

    private val token = "token"

    private val storeOrderRequestDTOMocks = StoreOrderRequestDTOMocks().storeOrderRequestDTOMocks

    private val storeId = 27L

    private val storePosDTOMock = StorePosDTOMock().storePosDTOs

    @Test
    fun test_Get_Configuration_Pos_By_Store_UseCase_invoke_when_service_returns_OK() {
        runBlocking {
            Mockito.`when`(
                configurationPosByStoreRepository.getPosRelatedToAStore(
                    storeId,
                    token
                )
            ).thenReturn(Result.Success(storePosDTOMock))

            val result = getConfigurationPosByStoreUseCase.invoke(
                storeId,
                token
            )
            Assertions.assertEquals(storePosDTOMock, result)
        }
    }

    @Test
    @Throws(Exception::class)
    fun test_Get_Configuration_Pos_By_Store_UseCase_invoke_when_service_returns_Error() {
        runBlocking {
            assertThrows<ResponseStatusException> {
                Mockito.`when`(
                    configurationPosByStoreRepository.getPosRelatedToAStore(
                        storeId,
                        token
                    )
                ).thenReturn(Result.Error(Exception("Some error"), HttpStatus.BAD_REQUEST))

                getConfigurationPosByStoreUseCase.invoke(
                    storeId,
                    token
                )
            }
        }
    }
}